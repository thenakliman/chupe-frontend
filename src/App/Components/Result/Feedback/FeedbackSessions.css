.feedbackSession-container {
  margin: 2rem 20rem;
}

.feedback-sessions-create-container {
  display: flex;
  justify-content: space-between;
  margin: 2rem auto;
}

.feedbackSession-input {
  height: 3rem;
  width: 48rem;
  font-size: 1.5rem;
}

.feedbackSession-button {
  padding: 0 1rem;
}