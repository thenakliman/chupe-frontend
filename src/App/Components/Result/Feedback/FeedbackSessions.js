import React from 'react';
import propTypes from 'prop-types';

require('./FeedbackSessions.css');

export class FeedbackSessions extends React.Component {
    componentDidMount() {
      this.props.getAllFeedbackSessions();
    }

    render() {
      return (
      <div id='feedbackSession-container-id'
           className='feedbackSession-container'>
          <div className='feedback-sessions-create-container'>
              <input id='feedbackSession-input-id'
                className='feedbackSession-input'
              />
              <button id='feedbackSession-button-id'
                className='feedbackSession-button'
                type='button'
              >
                Create Feedback Session
              </button>
          </div>
          <div>
            <table id='all-feedbackSessions-ordered-list'>
              <thead>
                <tr className='feedbackSession-table-header'>
                  <th> Id </th>
                  <th> description </th>
                  <th> Created By </th>
                </tr>
              </thead>
              <tbody id='all-feedbackSession-table-body-id'
                     className={'feedbackSession-table-body'}>
              {
                this.props.feedbackSessions.map((feedbackSession) => (
                  <tr key={`${feedbackSession.id}`}>
                    <td> {feedbackSession.id} </td>
                    <td className='feedbackSession-row'>
                            {feedbackSession.description}
                    </td>
                    <td> {feedbackSession.createdBy}</td>
                  </tr>
                ))
              }
              </tbody>
            </table>
          </div>
      </div>
      );
    }
}

FeedbackSessions.propTypes = {
  feedbackSessions: propTypes.array,
  getAllFeedbackSessions: propTypes.func.isRequired,
};
