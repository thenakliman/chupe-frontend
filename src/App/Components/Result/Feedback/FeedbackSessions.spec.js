/* eslint-disable */
import React from 'react';
import {FeedbackSessions} from './FeedbackSessions';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const feedbackSessions = [{
  description: 'retro - 1',
}, {
  description: 'retro - 2',
}];

describe('Show feedback sessions component', () => {
  it('should call getAllFeedbackSessions on component did mount ', () => {
    const getAllFeedbackSessions = jest.fn();
    shallow(
        <FeedbackSessions
          feedbackSessions={feedbackSessions}
          getAllFeedbackSessions={getAllFeedbackSessions}
        />
    );

    expect(getAllFeedbackSessions).toHaveBeenCalledWith();
  });

  it('should have rows for each feedback sessions', () => {
    const getAllFeedbackSessions = jest.fn();
    const wrapper = shallow(
        <FeedbackSessions
          feedbackSessions={feedbackSessions}
          getAllFeedbackSessions={getAllFeedbackSessions}
        />
    );

    expect(wrapper.find('tr').length).toBe(3);
  });

  it('should have input field for session create', () => {
    const getAllFeedbackSessions = jest.fn();
    const wrapper = shallow(
        <FeedbackSessions
          feedbackSessions={feedbackSessions}
          getAllFeedbackSessions={getAllFeedbackSessions}
        />
    );

    expect(wrapper.find('input').length).toBe(1);
  });

  it('should have input field for session create', () => {
    const getAllFeedbackSessions = jest.fn();
    const wrapper = shallow(
        <FeedbackSessions
          feedbackSessions={feedbackSessions}
          getAllFeedbackSessions={getAllFeedbackSessions}
        />
    );

    expect(wrapper.find('#feedbackSession-button-id').length).toBe(1);
  });

  describe('should match snapshot', () => {
    const wrapper = shallow(<FeedbackSessions
       feedbackSessions={feedbackSessions}
       getAllFeedbackSessions={jest.fn()}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('should have initial state with empty string', () => {
    const wrapper = shallow(<FeedbackSessions
       feedbackSessions={feedbackSessions}
       getAllFeedbackSessions={jest.fn()}
    />);

    expect(wrapper.state().description).toEqual('');
  });

  describe('should change internal state of the component', () => {
    const wrapper = shallow(<FeedbackSessions
       feedbackSessions={feedbackSessions}
       getAllFeedbackSessions={jest.fn()}
    />);

    const description = 'some description';
    wrapper.find('#feedbackSession-input-id')
        .simulate('change', {target: {value: description}});

    expect(wrapper.state().description).toEqual(description);
  });

  describe('should change internal state of the component', () => {
    const createFeedbackSession = jest.fn();
    const wrapper = shallow(<FeedbackSessions
       feedbackSessions={feedbackSessions}
       getAllFeedbackSessions={jest.fn()}
       createFeedbackSession={createFeedbackSession}
    />);

    const description = 'some description';
    wrapper.find('#feedbackSession-input-id')
        .simulate('change', {target: {value: description}});

    wrapper.find('#feedbackSession-button-id').simulate('click');

    expect(createFeedbackSession)
            .toHaveBeenCalledWith({description: description});
  });

  describe('should change state to empty string on create of session', () => {
    const createFeedbackSession = jest.fn();
    const wrapper = shallow(<FeedbackSessions
       feedbackSessions={feedbackSessions}
       getAllFeedbackSessions={jest.fn()}
       createFeedbackSession={createFeedbackSession}
    />);

    const description = 'some description';
    wrapper.find('#feedbackSession-input-id')
        .simulate('change', {target: {value: description}});

    wrapper.find('#feedbackSession-button-id').simulate('click');

    expect(wrapper.state().description).toBe('');
  });
});
