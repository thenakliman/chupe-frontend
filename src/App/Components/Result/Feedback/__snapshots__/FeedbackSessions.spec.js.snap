// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="feedbackSession-container"
  id="feedbackSession-container-id"
>
  <div
    className="feedback-sessions-create-container"
  >
    <input
      className="feedbackSession-input"
      id="feedbackSession-input-id"
      onChange={[Function]}
      value=""
    />
    <button
      className="feedbackSession-button"
      id="feedbackSession-button-id"
      onClick={[Function]}
      type="button"
    >
      Create Feedback Session
    </button>
  </div>
  <div>
    <table
      id="all-feedbackSessions-ordered-list"
    >
      <thead>
        <tr
          className="feedbackSession-table-header"
        >
          <th>
             Id 
          </th>
          <th>
             description 
          </th>
          <th>
             Created By 
          </th>
        </tr>
      </thead>
      <tbody
        className="feedbackSession-table-body"
        id="all-feedbackSession-table-body-id"
      >
        <tr
          className="selectable-row"
          id="feedback-sessions-10101"
          key="10101"
          onClick={[Function]}
        >
          <td>
             
            10101
             
          </td>
          <td
            className="feedbackSession-row"
          >
            retro - 1
          </td>
          <td>
             
          </td>
        </tr>
        <tr
          className="selectable-row"
          id="feedback-sessions-10102"
          key="10102"
          onClick={[Function]}
        >
          <td>
             
            10102
             
          </td>
          <td
            className="feedbackSession-row"
          >
            retro - 2
          </td>
          <td>
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
